prediction_market_agent/agents/abstract.py

```diff
@@ -1,8 +1,20 @@
+import typing as t
+from prediction_market_agent.data_models.market_data_models import MarketProtocol
+
+MarketVar = t.TypeVar("MarketVar", bound=MarketProtocol)
+
+
 class AbstractAgent:
     def __init__(self):
         pass
 
-    def run(self, objective: str) -> bool:
+    def pick_market(self, markets: list[MarketVar]) -> MarketVar:
```

Comment:
I added abstract pick_market for the infinite runs, but there is no implementation for this one yet.

Reply:
Yes! Updated.